modules/perception/obstacle/camera/visualizer/frame_content.cc

```diff
@@ -0,0 +1,392 @@
+            best_ts = it_ts;
+        }
+    }
+    _current_radar_timestamp = best_ts;
+    for (std::map<double, RadarContent>::iterator it = _radar_caches.begin();
+         it != _radar_caches.end();) {
+        if (it->first < best_ts) {
+            _radar_caches.erase(it++);
+        } else {
+            it++;
+        }
+    }
+    best_delta = FLT_MAX;
+    best_ts = -1;
+    for (std::map<double, FusionContent>::iterator it = _fusion_caches.begin();
+         it != _fusion_caches.end(); it++) {
+        double it_ts = it->first;
+        double delta = fabs(it_ts - ref);
+
+        if (delta < best_delta) {
+            best_delta = delta;
+            best_ts = it_ts;
+        }
+    }
+    _current_fusion_timestamp = best_ts;
+    for (std::map<double, FusionContent>::iterator it = _fusion_caches.begin();
+         it != _fusion_caches.end();) {
+        if (it->first < best_ts) {
+            _fusion_caches.erase(it++);
+        } else {
+            it++;
+        }
+    }
+
+    // find camera tracked best ts
+    best_delta = FLT_MAX;
+    best_ts = -1;
+    for (std::map<double, CameraContent>::iterator it = _camera_caches.begin();
+         it != _camera_caches.end(); it++) {
+        double it_ts = it->first;
+        double delta = fabs(it_ts - ref);
+
+        if (delta < best_delta) {
+            best_delta = delta;
+            best_ts = it_ts;
+        }
+    }
+    _current_camera_timestamp = best_ts;
+    for (std::map<double, CameraContent>::iterator it = _camera_caches.begin();
+         it != _camera_caches.end();) {
+        if (it->first < best_ts) {
+            _camera_caches.erase(it++);
+        } else {
+            it++;
+        }
+    }
+
+    best_delta = FLT_MAX;
+    best_ts = -1;
+    for (std::map<double, GroundTruthContent>::iterator it = _gt_caches.begin();
+         it != _gt_caches.end(); it++) {
+        double it_ts = it->first;
+        double delta = fabs(it_ts - ref);
+        if (delta < best_delta) {
+            best_delta = delta;
+            best_ts = it_ts;
+        }
+    }
+    _current_gt_timestamp = best_ts;
+    for (std::map<double, GroundTruthContent>::iterator it = _gt_caches.begin();
+         it != _gt_caches.end();) {
+        if (it->first < best_ts) {
+            _gt_caches.erase(it++);
+        } else {
+            it++;
+        }
+    }
+
+
+    // get motion timestamp
+    best_delta = FLT_MAX;
+    best_ts = -1;
+    for (std::map<double, MotionContent>::iterator it = _motion_caches.begin();
+         it != _motion_caches.end(); it++) {
+        double it_ts = it->first;
+        double delta = fabs(it_ts - ref);
+
+        if (delta < best_delta) {
+            best_delta = delta;
+            best_ts = it_ts;
+        }
+    }
+    _current_motion_timestamp = best_ts;
+    for (std::map<double, MotionContent>::iterator it = _motion_caches.begin();
+         it != _motion_caches.end();) {
+        if (it->first < best_ts) {
+            _motion_caches.erase(it++);
```

Comment:
need to change the format for private numbers

Reply:
will change later after code is stable